[TASK] Document ability to use distinct flashmessage queues (#5439)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/FlashMessages/Extbase.rst

Where to display the flash messages in an Extbase-based backend module is
as simple as moving the ViewHelper around.

By default, all messages are put into the scope of the

> the current plugin namespace with a prefix `extbase.flashmessages.`. So

Reviewer: ```suggestion
current plugin namespace with a prefix `extbase.flashmessages.`. So
```

if your plugin namespace is computed as `tx_myvendor_myplugin`, the
flash message queue identifier will be

> `extbase.flashmessages.tx_myvendor_myplugin`.

Reviewer 2: And where I have to set this identifier? How to inform flashMessage VH to use explicit this identifier?

Reviewer 3: Its a free identifier. And the example
below shows how to access it?!?

Reviewer 4: Here is only described the default, added more why and how in the section below

Author: Thanks!

> If you need distinct queues, you can use a custom identifier to fetch

Reviewer 2: Why should I need distinct queues?

Reviewer 3: If you have different levels of application logic. See the issue that georg linked for an example.

Reviewer 4: I added a headline an a paragraph about this

Where to display the flash messages in an Extbase-based backend module is
as simple as moving the ViewHelper around.

By default, all messages are put into the scope of the
current plugin namespace with a prefix `extbase.flashmessages.`. So
if your plugin namespace is computed as `tx_myvendor_myplugin`, the
flash message queue identifier will be
`extbase.flashmessages.tx_myvendor_myplugin`.

..  _flash-messages-extbase-distinct:

Using explicit flash message queues in Extbase
==============================================

It is possible to add a message to a different flash message queue. Use 

> cases would be detailed display of different Flash message queues in 

Author: ```suggestion
cases could be a detailed display of different flash message queues in 
```

> different places of the page  or displaying a Flash message when you 

Author: ```suggestion
different places of the page  or displaying a flash message when you 
```

forward to a different controller or even a different extension. 

If you need distinct queues, you can use a custom identifier to fetch
and operate on that queue:

..  code-block:: php

    $customQueue = $this->getFlashMessageQueue('tx_myvendor_customqueue');
    // Instead of using $this->addFlashMessage() you will instead directly
    // access the custom queue:
    $flashMessage = GeneralUtility::makeInstance(
            FlashMessage::class,
            'My flash message in a custom queue',
            'My flash message title of a custom queue',
            ContextualFeedbackSeverity::OK,
            $storeInSession = true,
    );
    $customQueue->enqueue($flashMessage);

..  _flash-messages-extbase-fluid-queueidentifier:

Fluid flash messages ViewHelper with explicit queue identifier
==============================================================

When you used an :ref:`explicit flash message queue <flash-messages-extbase-distinct>`

> during enqueueing the message it will only be displayed on the page if you use the

Author: ```suggestion
during enqueueing the message, it will only be displayed on the page if you use the
```
